kubectl-cloudflow/config/config_test.go

```diff
@@ -310,6 +311,291 @@ func Test_validateConfig(t *testing.T) {
+	assert.NotEmpty(t, validateConfig(badLabelKeyMalformed, spec))
+	fmt.Printf("badLabelKeyMalformed: %s\n", validateConfig(badLabelKeyMalformed, spec))
+
+	badLabelKeyMalformed2 := newConfig(`
+     cloudflow.runtimes.flink.kubernetes.pods.pod {
+		labels {
+		 	"lkjsdfsdf..sdfsfd//keyabcdefstuvzabcdefghijklmnopqrstuvwxyz" :  value2
+		 }
+	}
+	`)
+	assert.NotEmpty(t, validateConfig(badLabelKeyMalformed2, spec))
+	fmt.Printf("badLabelKeyMalformed2: %s\n", validateConfig(badLabelKeyMalformed2, spec))
+
+	badLabelKeyMalformed3 := newConfig(`
+		cloudflow.runtimes.flink.kubernetes.pods.pod {
+		  containers.container {
+		    resources {
+		      requests {
+		        cpu=2
+		        memory = "512M"
+		      }limits {
+		        memory = "1024M"
+		      }
+		    }
+		  }labels {
+		    "lkjsdfsdfsdfs+fd/keyabcdefstuvzabcdefghijklmnopqrstuvwxyz": value2
+		  }
+		}
+	`)
+	assert.NotEmpty(t, validateConfig(badLabelKeyMalformed3, spec))
+	fmt.Printf("badLabelKeyMalformed3: %s\n", validateConfig(badLabelKeyMalformed3, spec))
+
+	badLabelKeyPrefixMalformed := newConfig(`
+		cloudflow.runtimes.flink.kubernetes.pods.pod {
+		  containers.container {
+		    resources {
+		      requests {
+		        cpu=2
+		        memory = "512M"
+		      }limits {
+		        memory = "1024M"
+		      }
+		    }
+		  }labels {
+		    "0lkjsdfsdfsdfsfd/keyabcdefstuvzabcdefghijklmnopqrstuvwxyz": value2
+		  }
+		}
+	`)
+	assert.NotEmpty(t, validateConfig(badLabelKeyPrefixMalformed, spec))
+	fmt.Printf("badLabelKeyPrefixMalformed: %s\n", validateConfig(badLabelKeyPrefixMalformed, spec))
+
+	badLabelKeyPrefixTooLong := newConfig(`
+		cloudflow.runtimes.flink.kubernetes.pods.pod {
+		  containers.container {
+		    resources {
+		      requests {
+		        cpu=2
+		        memory = "512M"
+		      }limits {
+		        memory = "1024M"
+		      }
+		    }
+		  }labels {
+		    "flkjsdfsdfsdfsfdkeyabcdefstuvzabcdefghijklmnopqrstuvwxyzkeyghijklmnopqrstuvwxyzkeyghijklmnopqrstuvwxyzkeyghijklmnopqrstuvwxyzkeyghijklmnopqrstuvwxyzkeyghijklmnopqrstuvwxyzkeyabcdefstuvzabcdefghijklmnopqrstuvwxyzkeyabcdefstuvzabcdefghijklmnopqrstuvwxyzkeyabcdefstuvzabcdefghijklmnopqrstuvwxyz/keyabcdefstuvzabcdefghijklmnopqrstuvwxyz": value2
+		  }
+		}
+	`)
+	assert.NotEmpty(t, validateConfig(badLabelKeyPrefixTooLong, spec))
+	fmt.Printf("badLabelKeyPrefixTooLong: %s\n", validateConfig(badLabelKeyPrefixTooLong, spec))
+
+	labelKeyWellFormed2 := newConfig(`
+		cloudflow.runtimes.flink.kubernetes.pods.pod {
+		  containers.container {
+		    resources {
+		      requests {
+		        cpu = 2
+		        memory = "512M"
+		      }limits {
+		        memory = "1024M"
+		      }
+		    }
+		  }labels {
+		    "lkjsdfsdfsdfsfd/keyabcdefstuvzabcdefghijklmnopqrstuvwxyz": value2
+		  }
+		}	
+	`)
+	assert.Empty(t, validateConfig(labelKeyWellFormed2, spec))
+
+	labelWellFormed := newConfig(`
+     cloudflow.runtimes.flink.kubernetes.pods.pod {
+		labels {
+		 	keyabcdefstuv=jzabcdefghijklmnopqrstuvwxyz : value2
```

Comment:
why is this well-formed?

Reply:
:) just the name is well-formed. It actually raises the expected error

Reply:
changed

Reply:
aha! ok